### ISSUE TITLE
Kubernetes headless SSO

### ISSUE THREAD

**Issue by r0mant** [gravitational/teleport#30008]
In Teleport 12.2 we released headless SSO for SSH which allows users to use `tsh ssh` and `tsh scp` commands in headless mode on headless (browser-less) hosts which requires users to explicitly approve the "headless" request with MFA using CLI or web UI:

https://goteleport.com/docs/access-controls/guides/headless/

We want to extend this functionality to Kubernetes access so users can use `kubectl` in headless mode as well. We don't want to  require users to tap MFA key for every `kubectl` command as it would be poor UX (and wouldn't work for tools like Helm). The proposal is to introduce a headless mode for the Kube proxy instead so the flow would look like this:

- User starts Kube proxy in headless mode `tsh proxy kube --headless`.
- Before the proxy becomes functional and starts listening for client connections, the headless request needs to be approved with `tctl headless approve` or via web UI with the MFA tap like for SSH commands.
- Once it's approved, we configure the user's `kubectl` client to talk to this local proxy.
- To make sure other users on the machine can't talk to it, we should:
  - Add support for the proxy to listen on local Unix socket
  - Have some form of mutual TLS authentication between the user's kubectl and the proxy (e.g. a proxy can generate a self-signed CA, and configure user's local kubectl config with the client certificate signed by it).

Let's write and RFD before starting implementation.

**Comment by zmb3** [comment]
Duplicate of #24293?
